arviz/tests/test_data_pyro.py

```diff
@@ -48,6 +49,9 @@ def test_inference_data(self, data, eight_schools_params):
             "prior": ["mu", "tau", "eta"],
             "prior_predictive": ["obs"],
         }
+        if pyro.__version__.startswith("1"):

```

Comment:
What are the other pyro version where this won't work?

Reply:
I think what is happening is that pyro <1.0.0 (i.e. 0.5.1) does not store the observed data information. Therefore, even though `from_pyro` works in both versions, it will only contain observed data and log likelihood data if the version is >=1. Both versions need to be tested but using slightly different tests.

Reply:
Thank @OriolAbril! You have pointed out the whole story for this line of code.

Reply:
Hi, now reading again this, I think this actually needs a bit more robust check.

    import packaging

    ....

    packaging.version.parse(pyro.__version__) >= packaging.version.parse("1.0.0")

I mean I guess this will still work with 2.0 etc

Reply:
Thank @ahartikainen! I just made a change with your suggestion. It is better.